### PR TITLE
Update Network Connectivity Spoke Router and VPN Tunnels

### PR DESCRIPTION
**Pull request by r-teller**
- Remove immutable flag from VPN Tunnels to support:
  * includeImportRanges updates
- Remove immutable flag from Router Appliances to support:
  * Instance count updates
  * includeImportRanges updates
- Add correct update_mask_fields for VPN Tunnels and Router Appliances
- Fix Router Appliance instance attributes to be properly marked as required
- Update tests to verify:
  * Router Appliance scaling from 1 to 2 instances
  * Import range modifications
**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
networkconnectivity: added support for updating `linked_vpn_tunnels.include_import_ranges`, `linked_interconnect_attachments.include_import_ranges`, `linked_router_appliance_instances. instances` and `linked_router_appliance_instances.include_import_ranges` in `google_network_connectivity_spoke`
```
```release-note:bug
networkconnectivity: fixed `linked_router_appliance_instances.instances.virtual_machine` and `linked_router_appliance_instances.instances.ip_address` attributes in `google_network_connectivity_spoke` to be correctly marked as required. Otherwise the request to create the resource will fail.
```

### CODE REVIEW

**Comment by r-teller** [comment]
Before Change

```
 # module.usw1_appliance-vyoss.module.gcp_ncc_appliance_spoke_vyos["spoke-lan-vyos-transit-1ncc-usw1"].google_network_connectivity_spoke.ncc_spoke must be replaced
-/+ resource "google_network_connectivity_spoke" "ncc_spoke" {
      ~ create_time      = "2024-12-07T16:47:44.381551298Z" -> (known after apply)
      ~ hub              = "projects/rteller-demo-hst-e265-aaaa/locations/global/hubs/hub-lan-1ncc-gbl0" -> "hub-lan-1ncc-gbl0"
      ~ id               = "projects/rteller-demo-hst-e265-aaaa/locations/us-west1/spokes/spoke-lan-vyos-transit-1ncc-usw1" -> (known after apply)
      - labels           = {} -> null
      ~ name             = "projects/rteller-demo-hst-e265-aaaa/locations/us-west1/spokes/spoke-lan-vyos-transit-1ncc-usw1" -> "spoke-lan-vyos-transit-1ncc-usw1"
      ~ state            = "ACTIVE" -> (known after apply)
      ~ unique_id        = "c79f20f2-ae8d-46e0-98a5-addef5c406b6" -> (known after apply)
      ~ update_time      = "2024-12-07T16:56:41.257471932Z" -> (known after apply)
        # (5 unchanged attributes hidden)

      ~ linked_router_appliance_instances {
          - include_import_ranges      = [] -> null
            # (1 unchanged attribute hidden)

          - instances { # forces replacement
              - ip_address      = "172.16.0.23" -> null # forces replacement
              - virtual_machine = "https://www.googleapis.com/compute/v1/projects/rteller-demo-hst-e265-aaaa/zones/us-west1-a/instances/lan-vyos-transit-usw1-1ncc-01" -> null # forces replacement
            }

            # (1 unchanged block hidden)
        }
    }
```

After Change

```
  # module.usw1_appliance-vyoss.module.gcp_ncc_appliance_spoke_vyos["spoke-lan-vyos-transit-1ncc-usw1"].google_network_connectivity_spoke.ncc_spoke will be updated in-place      
  ~ resource "google_network_connectivity_spoke" "ncc_spoke" {
        id               = "projects/rteller-demo-hst-e265-aaaa/locations/us-west1/spokes/spoke-lan-vyos-transit-1ncc-usw1"
        name             = "projects/rteller-demo-hst-e265-aaaa/locations/us-west1/spokes/spoke-lan-vyos-transit-1ncc-usw1"
        # (11 unchanged attributes hidden)

      ~ linked_router_appliance_instances {
            # (2 unchanged attributes hidden)

          - instances {
              - ip_address      = "172.16.0.23" -> null
              - virtual_machine = "https://www.googleapis.com/compute/v1/projects/rteller-demo-hst-e265-aaaa/zones/us-west1-a/instances/lan-vyos-transit-usw1-1ncc-01" -> null    
            }

            # (1 unchanged block hidden)
        }
    }
```

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 1]
```diff
@@ -203,11 +209,13 @@ properties:
               type: String
               description: The URI of the virtual machine resource
               immutable: true
+              required: true
               diff_suppress_func: 'tpgresource.CompareSelfLinkOrResourceName'
             - name: 'ipAddress'
               type: String
               description: The IP address on the VM to use for peering.
               immutable: true
+              required: true
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:218]: Are these two fields required in the API Create request? I wonder if it is a breaking change for the users to mark the fields required.

Comment (robert-teller-sada) [reply mmv1/products/networkconnectivity/Spoke.yaml:218]: Yes these fields are currently required in the API

```
gcloud network-connectivity spokes linked-router-appliances create spoke-lan-vyos-transit-1ncc-usw1 --region us-west1 \
        --router-appliance=instance="https://www.googleapis.com/compute/v1/projects/rteller-demo-hst-e265-aaaa/zones/us-west1-a/instances/lan-vyos-transit-usw1-1ncc-01"
ERROR: (gcloud.network-connectivity.spokes.linked-router-appliances.create) argument --router-appliance: Key [ip] required in dict arg but not provided
Usage: gcloud network-connectivity spokes linked-router-appliances create (SPOKE : --region=REGION) --hub=HUB --router-appliance=[instance=INSTANCE],[ip=IP] [optional flags]
  optional flags may be  --async | --description | --group | --help |
                         --include-import-ranges | --labels | --region |
                         --site-to-site-data-transfer

For detailed information on this command and its flags, run:
  gcloud network-connectivity spokes linked-router-appliances create --help
```

Comment (r-teller) [reply mmv1/products/networkconnectivity/Spoke.yaml:218]: That is correct, this field is currently required in the API

https://cloud.google.com/network-connectivity/docs/reference/networkconnectivity/rest/Shared.Types/Spoke#routerapplianceinstance

```
 gcloud network-connectivity spokes linked-router-appliances create spoke-lan-vyos-transit-1ncc-usw1 --region us-west1 \
        --hub="projects/rteller-demo-hst-e265-aaaa/locations/global/hubs/hub-lan-1ncc-gbl0" \
        --router-appliance=instance="https://www.googleapis.com/compute/v1/projects/rteller-demo-hst-e265-aaaa/zones/us-west1-a/instances/lan-vyos-transit-usw1-1ncc-01"
ERROR: (gcloud.network-connectivity.spokes.linked-router-appliances.create) argument --router-appliance: Key [ip] required in dict arg but not provided
Usage: gcloud network-connectivity spokes linked-router-appliances create (SPOKE : --region=REGION) --hub=HUB --router-appliance=[instance=INSTANCE],[ip=IP] [optional flags]
  optional flags may be  --async | --description | --group | --help |
                         --include-import-ranges | --labels | --region |
                         --site-to-site-data-transfer

For detailed information on this command and its flags, run:
  gcloud network-connectivity spokes linked-router-appliances create --help
  
  ```

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 2]
```diff
@@ -131,6 +131,8 @@ properties:
       - linked_router_appliance_instances
       - linked_vpc_network
       - linked_producer_vpc_network
+    update_mask_fields:
+      - 'linkedVpnTunnels.includeImportRanges'
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:135]: Can you please add the update test for this field to this file https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_spoke_test.go?

This is the doc to add an update test https://googlecloudplatform.github.io/magic-modules/test/test/#add-an-update-test

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 3]
```diff
@@ -160,6 +162,8 @@ properties:
       - linked_router_appliance_instances
       - linked_vpc_network
       - linked_producer_vpc_network
+    update_mask_fields:
+      - 'linkedInterconnectAttachments.includeImportRanges'
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:166]: Can you please add the update test for this field to this file https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_spoke_test.go?

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/products/networkconnectivity/Spoke.yaml (hunk 2 above)
Comment (r-teller) [reply mmv1/products/networkconnectivity/Spoke.yaml:134]: tests have been added

File: mmv1/products/networkconnectivity/Spoke.yaml (hunk 3 above)
Comment (r-teller) [reply mmv1/products/networkconnectivity/Spoke.yaml:164]: tests have been added

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**